src/goTestExplorer.ts

```diff
@@ -37,6 +37,16 @@ export class GoTestExplorer {
         context.subscriptions.push(this.commands.testCompleted(this.onTestCompleted, this));
         testDiscoverer.discoverAllTests();
 
+        vscode.workspace.onDidSaveTextDocument(document => {
+            if (document.languageId !== 'go') {
+                return;
+            }
+            let testOnSave = vscode.workspace.getConfiguration('go')['testOnSave'];
+            if (!!testOnSave) {
+                vscode.commands.executeCommand('goTestExplorer.runAllTest');
```

Comment:
Thank you @kroppt  for raising the PR, I appreciate your efforts.
We can run the tests which belongs to the saved file only instead of running all the tests.
This extension group all the tests which belong a single file and call them **testSuite** and the same list is shown in the explorer. all the testSuite can be identified/retrieved by the [filepath and name](https://github.com/ppparihar/GoTestExplorer/blob/master/src/goTestProvider.ts#L84)  and I think the same details be be found from the *document* object of *onDidSaveTextDocument*. once we the testsuite node we can the [runTestSuite](https://github.com/ppparihar/GoTestExplorer/blob/master/src/goTestExplorer.ts#L48). I haven't test  it but I think it can solve our problem.

Reply:
Thanks, this helped me figure it out.